src/f3/snapshot.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019-2025 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+#[cfg(test)]
+mod tests;
+
+use crate::{rpc::f3::F3PowerEntry, utils::multihash::MultihashCode};
+use cid::Cid;
+use fvm_ipld_encoding::{IPLD_RAW, tuple::*};
+use integer_encoding::VarIntReader as _;
+use std::io::Read;
+
+pub fn get_f3_snapshot_cid(f3_data: &mut impl Read) -> anyhow::Result<Cid> {
+    Ok(Cid::new_v1(
+        IPLD_RAW,
+        MultihashCode::Blake2b256.digest_byte_stream(f3_data)?,
+    ))
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize_tuple, Deserialize_tuple)]
+pub struct F3SnapshotHeader {
+    pub version: u64,
+    pub first_instance: u64,
+    pub latest_instance: u64,
+    pub initial_power_table: Vec<F3PowerEntry>,
+}
```

Comment:
It'd be nice to have some docs around it.

Reply:
Fixed.